cfn-resources/teams/cmd/resource/resource.go

```diff
@@ -0,0 +1,432 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	progressevents "github.com/mongodb/mongodbatlas-cloudformation-resources/util/progressevent"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/constants"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/logger"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util/validator"
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+var CreateRequiredFields = []string{constants.PubKey, constants.PvtKey, constants.OrgID}
+var ReadRequiredFields = []string{constants.PubKey, constants.PvtKey, constants.OrgID, constants.TeamID}
+
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+
+	_, _ = logger.Debugf("Create Team - Request:%+v", currentModel)
+
+	// Validate required fields in the request
+	event, client, err := ValidateRequest(CreateRequiredFields, currentModel)
+	if err != nil {
+		return event, err
+	}
```

Comment:
`ValidateRequest` should only validate the request. it should not create the client and the event as well

```suggestion
	err := ValidateRequest(CreateRequiredFields, currentModel); err !=nill{
		return nill, err
	}
```